src/sentry/static/sentry/app/views/settings/projectAlerts/alertTypeChooser.tsx

```diff
@@ -137,6 +158,100 @@ const AlertTypeChooser = ({selected, onChange}: Props) => (
   </Container>
 );
 
+type State = {
+  type?: 'frequency' | 'issues';
+  granularity?: 'project' | 'issue';
+};
+
+class TypeChooserFlow extends React.Component<Props, State> {
+  state: State = {};
+
+  update = (state: Partial<State>) =>
+    this.setState(state, () => {
+      const {type, granularity} = this.state;
```

Comment:
I believe it's fine having this code here. I would only suggest moving it to another function and maybe use componentDidUpdate. In the componentDidUpdate you could do something like:

```
if (!isEqual(prevState, this.state)) {
this.yourMethod()
} 
```
I am trying to think more hooks, like " ok and if I was using useEffect here, how would it looks like?" but as I said...it's just a suggestion 😉